packages/autocomplete-js/src/__tests__/requester.test.ts

```diff
@@ -346,6 +350,93 @@ describe('requester', () => {
     });
   });
 
+  test('batches calls when possible and re-dispatches results to the right sources across requester instances', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+
+    const searchClient = createSearchClient({
+      search: jest.fn(() =>
+        Promise.resolve(
+          createMultiSearchResponse<{ label: string }>(
+            { hits: [{ objectID: '1', label: 'Hit 1' }] },
+            { hits: [{ objectID: '2', label: 'Hit 2' }] }
+          )
+        )
+      ),
+    });
+
+    const getResults1 = (params) =>
+      createRequester(fetchAlgoliaResults)({
+        transformResponse: (response) => response.hits,
+      })(params);
+
+    const getResults2 = (params) =>
+      createRequester(fetchAlgoliaResults)({
+        transformResponse: (response) => response.hits,
+      })(params);
```

Comment:
These requesters are considered similar because their `requesterId` is not passed, correct?

See previous comment but I'm not sure we should let requesters with `requesterId=undefined` be batched.